### PR TITLE
Refactored and made tests

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ✔️
| Tests pass?   | ✔️

### CODE REVIEW

File: src/Event/MissingTranslationEvent.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Yiisoft\I18n\Event;
+
+class MissingTranslationEvent
+{
+    public function __construct(string $category, string $language, string $message)
```
Comment (samdark) [inline src/Event/MissingTranslationEvent.php]: Looks like constructor is not doing anything. Was that intended?

File: src/Translator/TranslationsLoaderInterface.php [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Yiisoft\I18n\Translator;
+
+interface TranslationsLoaderInterface
```
Comment (samdark) [inline src/Translator/TranslationsLoaderInterface.php]: It is now https://github.com/yiisoft/i18n/blob/master/src/MessageReaderInterface.php#L6 Three separate methods were introduced to make loading delayed and allow loading only part of translations that is likely to save memory and network when having many translations in database.

File: src/Translator/Translator.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Yiisoft\I18n\Translator;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Yiisoft\I18n\Event\MissingTranslationEvent;
+use Yiisoft\I18n\TranslatorInterface;
+
+/**
+ * TODO may be it must be taken out from yiisoft/i18n?
```
Comment (samdark) [inline src/Translator/Translator.php]: No, it belongs to the package. Let's have it here.

File: src/Translator/TranslationsLoaderInterface.php (hunk 2 above)
Comment (xepozz) [reply src/Translator/TranslationsLoaderInterface.php]: Hm... okay. I think it needs to work with declensions.

What is `$context`?

File: src/Event/MissingTranslationEvent.php (hunk 1 above)
Comment (xepozz) [reply src/Event/MissingTranslationEvent.php]: I just kept original contructor.
 It needs to be implemented, but I didn't want to have to do this :)
May be later.

File: src/Translator/Translator.php (hunk 3 above)
Comment (xepozz) [reply src/Translator/Translator.php]: ok

File: src/Translator/TranslationsLoaderInterface.php (hunk 2 above)
Comment (samdark) [reply src/Translator/TranslationsLoaderInterface.php]: You mean plurals? Here you go: https://github.com/yiisoft/i18n/blob/master/src/MessageReaderInterface.php#L8

$context is where you pass parameters for the message. See https://www.yiiframework.com/doc/guide/2.0/en/tutorial-i18n#message-parameters

**Comment by xepozz** [comment]
No problem

**Comment by samdark** [comment]
Thanks!

**Pull request merged**